File: java/MASTER_SLAVE_WHOLE_PART_PROXY/GestorDatosCompra.java
public class GestorDatosCompra{
    //Atributos
    private static DatosCompra datosCompra = new DatosCompra();

    //Constructor
    public GestorDatosCompra(){}
    
    //Métodos
    static DatosCompra preparaDatosCompra(String numTarjeta, int diaC, int mesC, int anioC){
        datosCompra.setDatosCompra(numTarjeta, diaC, mesC, anioC);
        return datosCompra;
    }
}

File: php/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/gestores/GestorDatosCompra.php
<?php
require './datos/DatosCompra.php';

class GestorDatosCompra{
    //Atributos
    public static $datosCompra;

    public static function init(){
        self::$datosCompra = new DatosCompra(); 
    }

    public static function preparaDatosCompra($numTarjeta, $fecha){
        self::init();
        self::$datosCompra->setDatosCompra($numTarjeta, $fecha);
        return self::$datosCompra;
    }

}
